server/manifest.go

```diff
@@ -25,30 +29,59 @@ func (m *Manifest) Size() (size int64) {
 	return
 }
 
-func ParseNamedManifest(name model.Name) (*Manifest, error) {
-	if !name.IsFullyQualified() {
-		return nil, model.Unqualified(name)
+func (m *Manifest) Remove() error {
+	if err := os.Remove(m.filepath); err != nil {
+		return err
+	}
+
+	for _, layer := range append(m.Layers, m.Config) {
+		if err := layer.Remove(); err != nil {
+			return err
+		}
+	}
+
+	manifests, err := GetManifestPath()
+	if err != nil {
+		return err
+	}
+
+	return PruneDirectory(manifests)
+}
+
+func ParseNamedManifest(n model.Name) (*Manifest, error) {
+	if !n.IsFullyQualified() {
```

Comment:
What is Qualified in this case? Does this mean it has all its components (vs just `llama3`) - wouldn't that always be the case if we have defaults?

Reply:
In other words could we just catch this in `IsValid`?

Reply:
not necessarily. there's nothing stopping anyone from passing in an empty name

```go
ParseNamedManifest(model.Name{})
```